src/main/java/dev/sigstore/rekor/client/RekorClient.java

```diff
@@ -88,16 +91,70 @@ public RekorResponse putEntry(HashedRekordRequest hashedRekordRequest) throws IO
     if (resp.getStatusCode() != 201) {
       throw new IOException(
           String.format(
-              "bad response from rekor @ '%s' : %s", rekorEndpoint, resp.parseAsString()));
+              "bad response from rekor @ '%s' : %s", rekorPutEndpoint, resp.parseAsString()));
     }
 
     URI rekorEntryUri = serverUrl.resolve(resp.getHeaders().getLocation());
     String entry = resp.parseAsString();
     return RekorResponse.newRekorResponse(rekorEntryUri, entry);
   }
 
-  /** Obtain an entry for an artifact from rekor. */
-  public void getEntry() {
-    throw new UnsupportedOperationException("I'm a worthless download function");
+  public Optional<RekorEntry> getEntry(String UUID) throws IOException {
+    URI getEntryURI = serverUrl.resolve(REKOR_ENTRIES_PATH + "/" + UUID);
+    HttpRequest req =
+        httpProvider
+            .getHttpTransport()
+            .createRequestFactory()
+            .buildGetRequest(new GenericUrl(getEntryURI));
+    req.getHeaders().set("Accept", "application/json");
+    HttpResponse response;
+    try {
+      response = req.execute();
+    } catch (HttpResponseException e) {
+      if (e.getStatusCode() == 404) return Optional.ofNullable(null);
```

Comment:
404 is explicitly for resource not found. I believe it's the standard response for any restful services to indicate absence rather than any error. That being said I don't feel strongly either way, I'm leaning towards this approach based off exceptions typically being reserved for error conditions and not expected control flow. For instance that's why normal map behaviour is to return null for key look ups that aren't present.

Reply:
Yeah you're right, I was thinking of 500 for some reason. nvm.